spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -204,15 +223,26 @@ private void writeAndValidateWithLocations(Table table, File location, File expe
   }
 
   private List<Row> readTable(String location) {
-    Dataset<Row> result = spark.read().format("iceberg").load(location);
+    return readTable(location, SnapshotRef.MAIN_BRANCH);
+  }
+
+  private List<Row> readTable(String location, String branch) {
+    Dataset<Row> result =
+        spark.read().format("iceberg").option(SparkReadOptions.BRANCH, branch).load(location);
```

Comment:
Minor: In tests, I'd generally prefer not setting the branch option when we intend to write to main. Otherwise we're not testing the default case. I know that it is currently equivalent, but it seems like a gap that could eventually introduce errors if all of our tests use a specific branch.

Reply:
Good point, is it alright if I get this in a follow on PR before release?